fhir-server/src/main/java/com/ibm/fhir/server/operation/spi/AbstractOperation.java

```diff
@@ -143,6 +143,10 @@ protected void validateOperationContext(FHIROperationContext operationContext, C
                 String msg = "Operation context RESOURCE_TYPE is not allowed for operation: '" + getName() + "'";
                 throw buildExceptionWithIssue(msg, IssueType.INVALID);
             } else {
+                if (resourceType == null) {
+                    String msg = "Operation context RESOURCE_TYPE must include a valid ResourceType in the URL Path";
+                    throw buildExceptionWithIssue(msg, IssueType.INVALID);
```

Comment:
I'm thinking it should maybe be a not-supported (mapped to a 404) instead of an invalid (mapped to 400).

That would be more consistent with what we return if you try to read/create/update/delete a resource for an invalid resource type.  For example, this is our 404 response for a `GET [base]/Fake/1`
```
{
    "resourceType": "OperationOutcome",
    "id": "ac-1d-0-3-fda86c56-02dc-4ce9-970b-080203c030e4",
    "issue": [
        {
            "severity": "fatal",
            "code": "not-supported",
            "_code": {
                "extension": [
                    {
                        "url": "http://ibm.com/fhir/extension/not-supported-detail",
                        "valueCode": "resource"
                    }
                ]
            },
            "details": {
                "text": "'Fake' is not a valid resource type."
            }
        }
    ]
}
```

Reply:
actually, i take it back.  we do a 404 for READ, but not for create.  interesting...

Reply:
I'd prefer not to pass back the resource type since that'd mutate the signature for invoke pretty heavily, unless I pack it in the FHIROperationContext.

Reply:
```